### ISSUE TITLE
Implement working Task that deploys to k8s using helm

### ISSUE THREAD

**Issue by bobcatfish** [knative/build-pipeline#63]
## Expected Behavior
Once we have implemented a simple hellworld Task (#59) a user should be able to define a Task that can deploy at a k8s resource using [helm](https://helm.sh/).

This would mean that:
* Templating is in place that allows for the Task to use the image and registry values specified in the input [image resource](https://github.com/knative/build-pipeline/blob/master/pkg/apis/pipeline/v1beta1/image_source.go) in the steps of the Task as arguments to the helm commands
* Any required service account needed for accessing to the cluster is available to the Task
* Any logic needed to support the cluster type (see #60) is implemented

To try this out manually, you'll need a k8s cluster has been setup that you can deploy to.

Should include:
* Expanding the docs about how to write a Task that uses an k8s cluster resource, and how to set up the service account correctly
* Add docs on how to debug this when it doesn't work
* Test coverage, unit and also integration if possible (see #16)

## Actual Behavior
We have no logic currently for even running a task, but #59 will add basic logic. #62 will expand on this but shouldn't block this task.

## Steps to Reproduce the Problem

  1. Deploy the Pipeline CRD + related CRDs to your cluster (note: should be docs on how to do this post kubebuilder)
  2. Create a Resource like this (you can use a public image such as the kritis one in this example or you can use any other image):
```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: Resource
metadata:
  name: helm-resources
  namespace: default
spec:
  resources:
    - name: kritis-app-github-resource
      type: github
      params:
      - name: url
        value: https://github.com/grafeas/kritis
      - name: revision
        value: master
    - name: kritis-server
      type: image
      params:
        - name: url
          value: gcr.io/kritis-project
```
  3. Create a Task like this:
```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: Task
metadata:
  name: helm
  namespace: default
spec:
    inputs:
        resources:
            - name: workspace
              type: github
            - name: imageToDeploy
              type: image
        params:
            - name: pathToHelmCharts
              value: string
            - name: helmArgs
              value: string
        cluster:
            - name: clusterToDeployTo
    buildSpec:
        steps:
        - name: deploy
          image: kubernetes-helm
          args: ['deploy', '--path=${pathToHelmChart}', '--set image=${imageToDeploy.Url}.${imageToDeploy.Name}', '${helmArgs}', '--cluster=${clusterToDeployTo.Endpoint}']
```
(Note I am pretty much making up the `helm` command here, please replace it with something that actually works and makes sense!)

  4. Create a `TaskRun` like this:

```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: TaskRun
metadata:
  name: helm-run-0
spec:
    taskRef:
        name: helm
    resourceRef:
       name: helm-resources
    trigger:
        triggerRef:
            type: manual
```
5. Look at the logs from the running container (Note the best docs we have on this at the moment are [some of the steps in the Build quick start](https://github.com/knative/docs/blob/master/build/creating-builds.md#creating-and-running-a-build)) to see that the helm command was successful (or not successful)
  6. Observe that the `status` of the `TaskRun` has been updated appropriately, e.g. see [this example](https://github.com/knative/build-pipeline/blob/1c4deea8de70d40ded6cab8d64fe303d65ce2970/examples/invocations/run-kritis-test.yaml#L37) which contains:
* Outcomes of individual steps
* Started, Completed, and Successful conditions
7. Look at the cluster itself and verify that the image has been deployed to it

## Additional Info
* Blocked by #59 (helloworld task), #58 (GitHub resource).
* Whether or not you want the cluster to actually be managed by helm (e.g. [as in this tutorial](https://github.com/grafeas/kritis/blob/master/install.md#step-6-install-and-configure-helm)) or just use helm to deploy is optional.
* [Kritis helm charts](https://github.com/grafeas/kritis/blob/master/kritis-charts/values.yaml), docs are [here](https://github.com/grafeas/kritis/blob/master/install.md#step-6-install-and-configure-helm)

**Comment by pivotal-nader-ziada** [comment]
Should we consider this as a `resource` since it is the output of a `Task`?

**Comment by bobcatfish** [comment]
> Should we consider this as a `resource` since it is the output of a `Task`?

By "this" do you mean the cluster than gets deployed to? Definitely!

Or do you mean the deployment itself? b/c that's an interesting idea that I don't think we'd thought of!

**Comment by pivotal-nader-ziada** [comment]
I meant the `deployment` of the chart, a `helm` resource that knows how to deploy similar to how the `image` resource knows how to upload the image to a registry

**Comment by bobcatfish** [comment]
> I meant the `deployment` of the chart, a `helm` resource that knows how to deploy similar to how the image resource knows how to upload the image to a registry

Oooo neat! I think that's a cool idea, and makes a lot of sense since deployments will probably be a big part of what I imagine folks would want to use Pipelines for. Would you be interested in throwing together an issue to look into designing a `Deployment` resource?

**Comment by pivotal-nader-ziada** [comment]
Sure, I'll write up an issue for it

**Comment by ImJasonH** [comment]
If it helps, Helm's founder contributed and maintains a container image that runs `helm`, with GKE integration, [here](https://github.com/GoogleCloudPlatform/cloud-builders-community/tree/master/helm).

**Comment by pivotal-nader-ziada** [comment]
/close
addressed by PR: #160 
Will address the cluster selection by another issue and pr

**Comment by bobcatfish** [comment]
Okay I created #177 to add support for clusters, closing this issue.
